code/modules/reagents/chemistry/recipes/pyrotechnics.dm

```diff
@@ -471,10 +472,21 @@
 
 /datum/chemical_reaction/reagent_explosion/nitrous_oxide
 	required_reagents = list(/datum/reagent/nitrous_oxide = 1)
-	strengthdiv = 7
+	strengthdiv = 9
 	required_temp = 575
 	modifier = 1
 
+/datum/chemical_reaction/reagent_explosion/nitrous_oxide/on_reaction(datum/reagents/holder, created_volume)
+	holder.remove_reagent(/datum/reagent/sorium, created_volume*2)
+	var/turf/T = get_turf(holder.my_atom)
+	//generally twice as weak as sorium.
+	var/range = clamp(sqrt(created_volume*2), 1, 6)
+	//This first throws people away and then it explodes
+	goonchem_vortex(T, 1, range)
+	T.atmos_spawn_air("o2=[created_volume/2];TEMP=[575]")
+	T.atmos_spawn_air("n2=[created_volume/2];TEMP=[575]")
```

Comment:
1 after spawning air always use `air_update_turf()` to let the turf know you spawned some gases and do its fancy stuff (in your case `T.air_update_turf()`)
2 how much air can you make with this? is it capped? and if so how much is the cap? because that's what messed up showers last time

Reply:
it will not mess up since it is an explosion ghilker. If you put too much n2o there it will just delete the gas since it will immidiately waste to space. so you would have to do constant very-low volume explosions. which just isnt feasable. air_update is already called when the explosion happens, no need of doing it twice.